icu4c/source/i18n/number_longnames.cpp

```diff
@@ -1454,6 +1469,39 @@ void LongNameHandler::processPatternTimes(MeasureUnitImpl &&productUnit,
             }
         }
     }
+
+    // 5. Handling constant denominator if it exists.
+    if (productUnit.constantDenominator != 0) {
+        int32_t pluralIndex = -1;
+        for (int32_t index = 0; index < StandardPlural::Form::COUNT; index++) {
+            if (!outArray[index].isBogus()) {
+                pluralIndex = index;
+                break;
+            }
+        }
+
+        U_ASSERT(pluralIndex >= 0); // "No plural form found for constant denominator"
+
+        // TODO(ICU-23039):
+        // Improve the handling of constant_denominator representation.
+        // For instance, a constant_denominator of 1000000 should be adaptable to
+        // formats like
+        // 1,000,000, 1e6, or 1 million.
+        // Furthermore, ensure consistent pluralization rules for units. For example,
+        // "meter per 100 seconds" should be evaluated for correct singular/plural
+        // usage: "second" or "seconds"?
+        // Similarly, "kilogram per 1000 meters" should be checked for "meter" or
+        // "meters"?
```

Comment:
Thanks for the TODO here.  And if it was there all along, I'm sorry I didn't notice it and put two and two together before I added my previous comments.

Reply:
No worries, it was there from the first version , but no problem :)